src/basic/process-util.c

```diff
@@ -1732,6 +1733,51 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, pid_t *ret_pid) {
+        posix_spawnattr_t attr;
+        sigset_t mask;
+        pid_t pid;
+        int r;
+
+        /* Forks and invokes 'path' with 'argv' and 'envp' using CLONE_VM and CLONE_VFORK, which means the
+         * caller will be blocked until the child either exits or exec's. The memory of the child will be
+         * fully shared with the memory of the parent, so that there are no copy-on-write or memory.max
+         * issues. */
+
+        assert(path);
+        assert(argv);
+        assert(ret_pid);
+
+        assert_se(sigemptyset(&mask) >= 0);
+
+        r = posix_spawnattr_init(&attr);
+        if (r != 0)
+                return -r; /* These functions return a positive errno on failure */
+        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGMASK);
+        if (r != 0)
+                goto error;
+        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGDEF);
+        if (r != 0)
+                goto error;
+        r = posix_spawnattr_setsigmask(&attr, &mask);
```

Comment:
set all bits in the sigse_t first, via sigfillset(), otherwise this is a nop

Reply:
isn't that done by `sigemptyset()` already?

Reply:
sigemptyset → all bits to zero
sigfillset → all bits to one

Reply:
got it - will switch to sigfillset instead of sigemptyset